src/parser.rs

```diff
@@ -1613,7 +1613,13 @@ impl<'a> Parser<'a> {
             Token::Mod => Some(BinaryOperator::Modulo),
             Token::StringConcat => Some(BinaryOperator::StringConcat),
             Token::Pipe => Some(BinaryOperator::BitwiseOr),
-            Token::Caret => Some(BinaryOperator::BitwiseXor),
+            Token::Caret => {
+                if dialect_of!(self is PostgreSqlDialect) {
```

Comment:
I've seen @AugustoFKL @alamb request documentation links elsewhere — it may be worthwhile to include one here.